### ISSUE TITLE
PoolVector3Array causes memory leaks in C++

### ISSUE THREAD

**Issue by roscale** [godotengine/godot#42559]
**Godot version:**

3.1.2

**OS/device including version:**

Arch Linux, Intel UHD 420, Mesa, GLES3

**Issue description:**

`PoolVector3Array` is not cleaning up the memory on destruction, causing a memory leak. When enough memory is leaked, the program crashes with the error message
```
ERROR: resize: All memory pool allocations are in use.
   At: ./core/pool_vector.h:524.
```

**Steps to reproduce:**
```C++
Array Library::function(Ref<Mesh> mesh) {
	Array mesh_array = mesh->surface_get_arrays(0);

	PoolVector3Array vertices = mesh_array[ArrayMesh::ARRAY_VERTEX];
	PoolVector3Array normals = mesh_array[ArrayMesh::ARRAY_NORMAL];
	PoolIntArray indices = mesh_array[ArrayMesh::ARRAY_INDEX];
        return Array();
}

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/18094.

**Comment by akien-mga** [comment]
Can you try with Godot 3.2.3? And ideally provide a project we can use to reproduce the issue.

**Comment by roscale** [comment]
I can't update to Godot 3.2 for other reasons but I tried compiling the library with godot-cpp 3.2 and it apparently the leak is gone. Do still need to provide a project or can I close this issue?

**Comment by akien-mga** [comment]
I gues this can be closed then, I don't have plans for a 3.1.3 release for now (might change in the future if there's strong demand).
